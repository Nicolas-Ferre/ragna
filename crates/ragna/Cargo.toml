[package]
name = "ragna"
description = "Library for easily creating GPU-native applications"
readme = "../../README.md"
keywords = ["gpu", "rendering", "compute", "shader"]
categories = ["game-development", "graphics", "rendering", "concurrency"]
exclude = [".github", "README.md"]
version.workspace = true
authors.workspace = true
edition.workspace = true
license.workspace = true
repository.workspace = true
rust-version.workspace = true

[dependencies]
bitfield-struct.workspace = true
derive-where.workspace = true
futures.workspace = true
fxhash.workspace = true
itertools.workspace = true
once_cell.workspace = true
wgpu.workspace = true
ragna_derive.workspace = true

[lints]
workspace = true
