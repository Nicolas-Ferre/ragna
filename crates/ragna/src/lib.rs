//! A library for creating easily a GPU-native application.

mod app;
mod operations;
mod operators;
mod runner;
mod types;
mod wgsl;

pub use app::*;
pub use operators::*;
pub use types::*;

/// Transforms a Rust module to a GPU module.
pub use ragna_derive::gpu;

// TODO: adapt compilation tests
