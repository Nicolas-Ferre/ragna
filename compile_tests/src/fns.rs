fn main() {}

#[ragna::gpu]
mod gpu {
    use ragna::{F32, U32};

    fn func_with_self(self) {}
}
