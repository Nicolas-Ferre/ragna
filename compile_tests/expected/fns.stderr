error: unsupported parameter
 --> compile_tests/src/fns.rs:7:23
  |
7 |     fn func_with_self(self) {}
  |                       ^^^^

error: `self` parameter is only allowed in associated functions
 --> compile_tests/src/fns.rs:7:23
  |
7 |     fn func_with_self(self) {}
  |                       ^^^^ not semantically valid as function parameter
  |
  = note: associated functions are those in `impl` or `trait` definitions


